models/models.go

```diff
@@ -0,0 +1,72 @@
+package models
+
```

Comment:
The implementation definitely has the marks of a bottom-up approach. It made sense at the time since this is  the "total" image of the system/platform that was provided in the description. Assumption could be made regarding the design and integration with other possible components but assumptions also have a 50/50 chance to increase the delta with the possible actual request. This design makes sense in the worldview that it is somewhat a standalone (as much as it can be) component, but in a bigger architecture with possibly many components, microservices, etc... total or partial redesign/extention/modification is certainly possible.